snuba/web/rpc/proto_component.py

```diff
@@ -0,0 +1,150 @@
+                visitor
+            )
+            TimeSeriesExpressionWrapper(time_series_expression.formula.right).accept(
+                visitor
+            )
+
+
+class TraceItemTableRequestWrapper(ProtoWrapper):
+    def accept(self, visitor: ProtoVisitor) -> None:
+        visitor.visit_TraceItemTableRequestWrapper(self)
+        trace_item_table_request = self.underlying_proto
+        for col in trace_item_table_request.columns:
+            ColumnWrapper(col).accept(visitor)
+        for ob in trace_item_table_request.order_by:
+            ColumnWrapper(ob.column).accept(visitor)
+        if trace_item_table_request.HasField("aggregation_filter"):
+            AggregationFilterWrapper(
+                trace_item_table_request.aggregation_filter
+            ).accept(visitor)
+
+
+class TimeSeriesRequestWrapper(ProtoWrapper):
+    def accept(self, visitor: ProtoVisitor) -> None:
+        visitor.visit_TimeSeriesRequestWrapper(self)
+        time_series_request = self.underlying_proto
+        for expression in time_series_request.expressions:
+            TimeSeriesExpressionWrapper(expression).accept(visitor)
+
+
+class ProtoVisitor(ABC):
+    def __getattr__(self, visit_method: str) -> Callable[..., Any]:
+        subclass_visit_method = getattr(self.__class__, visit_method, None)
+        if callable(subclass_visit_method):
+            return MethodType(subclass_visit_method, self)
+
+        def do_nothing(*args: Any, **kwargs: Any) -> None:
+            pass
+
+        return do_nothing
+
+
+def _convert_aggregation_to_conditional_aggregation(
+    input: Column | AggregationComparisonFilter | TimeSeriesExpression,
+) -> None:
+    if input.HasField("aggregation"):
+        aggregation = input.aggregation
+        input.ClearField("aggregation")
+        input.conditional_aggregation.CopyFrom(
+            AttributeConditionalAggregation(
+                aggregate=aggregation.aggregate,
+                key=aggregation.key,
+                label=aggregation.label,
+                extrapolation_mode=aggregation.extrapolation_mode,
+            )
+        )
+
+
+class AggregationToConditionalAggregationVisitor(ProtoVisitor):
+    """
+    We support aggregation, but now we want to support conditional aggregation for the insights team, which only aggregates
+    if the field satisfies the condition: https://clickhouse.com/docs/en/sql-reference/aggregate-functions/combinators#-if.
+    For messages that don't have conditional aggregation, this function replaces the aggregation with a conditional aggregation,
+    where the filter is null, and every field is the same. This allows code elsewhere to set the default condition to always
+    be true.
+    The reason we do this "transformation" is to avoid code fragmentation down the line, where we constantly have to check
+    if the request contains `AttributeAggregation` or `AttributeConditionalAggregation`
```

Comment:
This is not communicating the important parts of why this transformation exists. It can be summed up as:

Originally, the protobuf request did not support conditional aggregations. When they were added, the file structure did not allow adding a TraceItemFilter to the `AttributeAggregation` message without breaking backwards compatibility. A new `AttributeConditionalAggregation` message was added whose fields are a superset of of `AttributeAggregation`. `AttributeAggregation` was deprecated.

This code transforms every `AttributeAggregation` to an `AttributeConditionalAggregation` to be able to serve  queries before the deprecation, and have the rest of the query pipeline only concerned with one kind of aggregation object.